python_modules/dagster/dagster_tests/declarative_automation_tests/automation_condition_tests/builtins/test_data_version_changed.py

```diff
@@ -1,78 +1,113 @@
+        dg.AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "a"}, partition=partition_key)
     )
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # no longer "newly updated"
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 0
 
     # now updates with the same data version
     instance.report_runless_asset_event(
-        AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "a"}, partition=partition_key)
+        dg.AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "a"}, partition=partition_key)
     )
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 0
 
     # again
     instance.report_runless_asset_event(
-        AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "a"}, partition=partition_key)
+        dg.AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "a"}, partition=partition_key)
     )
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 0
 
     # new data version
     instance.report_runless_asset_event(
-        AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "b"}, partition=partition_key)
+        dg.AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "b"}, partition=partition_key)
     )
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # new data version
     instance.report_runless_asset_event(
-        AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "c"}, partition=partition_key)
+        dg.AssetMaterialization(foo.key, tags={DATA_VERSION_TAG: "c"}, partition=partition_key)
     )
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
     assert result.total_requested == 1
 
     # no longer "newly updated"
-    result = evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    result = dg.evaluate_automation_conditions(defs=defs, instance=instance, cursor=result.cursor)
+    assert result.total_requested == 0
+
+
+def test_data_version_changed_condition_with_external_asset_observations() -> None:
```

Comment:
feels like we should be detecting "mixed regime" here no? IE; new materialization event comes in after an observation event, what happens? Is it possible for an AssetMaterialization to have data version information attached? (Given that it boils down to a tag I assume yes)